Swamp Sneak/Assets/Scripts/rkScripts/GameControl.cs

```diff
@@ -144,10 +166,66 @@ private void updateSaveFile()
         writer.Close();
 
         //Re-import the file to update the reference in the editor
-     //   AssetDatabase.ImportAsset(path);
-     //   TextAsset asset = Resources.Load("save.txt");
+     	//AssetDatabase.ImportAsset(path);
+     	//TextAsset asset = Resources.Load("save.txt");
 
         //Print the text from the file
-      //  Debug.Log(asset.text);
+      	//Debug.Log(asset.text);
+
     }
-}
+
+	// Saves data out to a binary file. This function can be called anywhere the GameControl 
+	// object is accessible (which should be everywhere)
+	public void SaveData() {
+		string path = save_file_location + save_file_name;
+		BinaryFormatter bf = new BinaryFormatter ();
+
+		// Fill out a PersistentData struct
+		PersistentData save_data = new PersistentData();
+		save_data.player_level = player_character.getLevel ();
+		save_data.player_experience = player_character.getExperience ();
+		save_data.player_stealth = player_character.getStealth ();
```

Comment:
There should probably be a check for if these are `null`, so we don't essentially overwrite the save with nothing. Though that should never be the case if we are careful about when we call `SaveData`.

Reply:
Also a good idea! :) That was actually happening when I was initally calling SaveData in OnDisable, which was a bad idea, so I took it out.

Reply:
It looks like I actually already do a null check (if (player_character != null) ) ? I could do further checks for each individual stat, but those get initialized in Player's Start() function, and I can't imagine us ever wanting to call SaveData in any of the Start functions

Reply:
I don't see that check anywhere? Yeah checking each stat is prob overkill and unnecessary like you say.